addons/timecop/bootstrap.js

```diff
@@ -0,0 +1,40 @@
+const {classes: Cc, interfaces: Ci, utils: Cu} = Components;
+Cu.import("resource://gre/modules/Services.jsm");
+
+const BROWSER_SEARCH_PREF = "browser.search.";
+
+let gInstall = false;
+
+function install(aData, aReason) {
+  // We check for install here instead of in startup because
+  // In Firefox 46, the install reason didn't work properly
+  // in startup in some cases.
+  if (aReason != 5 /* ADDON_INSTALL */) {
+    return;
+  }
+  gInstall = true;
+}
+
+function uninstall(aData, aReason) {}
+
+function startup(aData, aReason) {
+  if (!gInstall) {
+    return;
+  }
+  Services.prefs.setBoolPref(BROWSER_SEARCH_PREF + "reset.geo", true);
+}
+
+function shutdown(aData, aReason) {
+  if (aReason != 2 /* APP_SHUTDOWN */) {
+    return;
+  }
+  try {
+    let geoReset = Services.prefs.getBoolPref(BROWSER_SEARCH_PREF + "reset.geo", false);
+    if (geoReset) {
+      Services.prefs.clearUserPref(BROWSER_SEARCH_PREF + "region");
+      Services.prefs.clearUserPref(BROWSER_SEARCH_PREF + "countryCode");
+      Services.prefs.clearUserPref(BROWSER_SEARCH_PREF + "isUS");
+      Services.prefs.clearUserPref(BROWSER_SEARCH_PREF + "reset.geo");
+    }
+  } catch (e) {}
```

Comment:
Worth sending telemetry?

Reply:
Might be worth sending telemetry for both success and failure case... are these prefs among the few we send along with Telemetry? Otherwise I can't think how we'd confirm that this worked or not.

Reply:
We're going to be doing the reset for everyone, so the only interesting thing would be if the new version was different than the old it would be a lot more work to find that out.